source/_posts/2015-10-11-a-configuration-management-rosetta-stone-configuring-sensu-with-puppet.markdown

```diff
@@ -0,0 +1,352 @@
+"Chef". This gives a lot of credibility to the Cookbook, and shows in the
+[contributor page](https://github.com/sensu/sensu-chef/graphs/contributors).
+
+The Cookbook itself is feature complete, with recipes for installing and
+configuring all aspects of Sensu.
+
+The scope of the cookbook includes all Sensu related technologies, including
+RabbitMQ, Redis, and Uchiwa. It is certainly "batteries included" and on by
+default.  It even downloads and compiles [Redis from
+source](https://github.com/sensu/sensu-chef/issues/340) for you.
+
+Another example of this is the RabbitMQ module
+[setting](https://github.com/jjasghar/rabbitmq/pull/301)
+[Apt](https://github.com/jjasghar/rabbitmq/pull/302/files) attributes. Like the
+above Redis example, this behavior surprised me, but technically not related to
+the Sensu chef cookbook.
+
+At the same time, [wrapper
+cookbooks](https://github.com/portertech/chef-monitor) are recommended as a
+method to combine multiple cookbooks together in a coherent way. I think in
+general I just expected the wrapper cookbooks to do more and the main Sensu
+cookbook to do less.
+
+The cookbook does have an integration test suite, but it is not run via Travis.
+The code is under active development, and does multiple releases a year. It has
+native support for Chef [data bags](https://docs.chef.io/data_bags.html) for
+transporting the RabbitMQ SSL support, which is a nice touch. (Not tested in
+this review)
+
+### Ansible
+
+#### Ansible in General
+
+Ansible is a relative new-comer to the configuration-management space. Ansible
+uses YAML files to define desired state. The yamls files are a nice way to
+represent things, but it would be misleading to think that Ansible is just yaml
+files. Ansible has its own DSL and uses Jinja2 templating, which is parsed
+*over* the contents of the yaml.
+
+The [Ansible Galaxy](https://galaxy.ansible.com/) is the community registry for
+uploading shared roles. You can [sort by rating](https://galaxy.ansible.com/list#/roles?page=1&per_page=10&sort_order=owner__username,name&f=apache)
+to try to get a better idea about which roles are potentially higher quality
+than others.
+
+There doesn't seem to be any official roles/playbooks? The closest there are to
+official roles is the
+[ansible-examples](https://github.com/ansible/ansible-examples/tree/master/lamp_simple)
+repository. But click the link and look at the `lamp_simple` example. There is
+no code-reuse at all! Every example re-invents how to install apache, install
+ntp, configure iptables, etc. What is up with that?
+
+While the yaml files may make it very easy for beginners to make playbooks that
+get things done quickly, I don't think they will work out great as
+infrastructure really expands. The [abstractions just are not there](https://github.com/ansible/ansible-examples/blob/49708518fab943a37bdbf15ee61177155f0cc50f/lamp_simple/roles/web/tasks/install_httpd.yml#L15-L17)
```

Comment:
I think 'really' doesn't pull its weight in this sentence.  'greatly' would be better, and there may be even stronger ways to get your point across.